ci/pipeline/test_rules.txt

```diff
@@ -87,7 +88,7 @@ python/ray/experimental/channel/
 ;
 
 python/
-@ ml tune train data
+@ data
```

Comment:
Thinking about this more, I think we should be a bit more conservative and start by only making the tests conditional on the C++ files.

Once I've removed dependencies on `ray._private` from the libraries, then I'd be comfortable making this change as well

Reply:
There are still dependencies on `ray._private.test_utils` and things like that